Userland/Libraries/LibGfx/ImageFormats/JBIG2Loader.cpp

```diff
@@ -1614,13 +1611,69 @@ static ErrorOr<NonnullOwnPtr<BilevelImage>> generic_refinement_region_decoding_p
 
     auto compute_context = inputs.gr_template == 0 ? compute_context_0 : compute_context_1;
 
+    // Figure 14 – Reused context for coding the SLTP value when GRTEMPLATE is 0
+    constexpr u16 sltp_context_for_template_0 = 0b000'010'000'000'0;
+
+    // Figure 15 – Reused context for coding the SLTP value when GRTEMPLATE is 1
+    constexpr u16 sltp_context_for_template_1 = 0b0'010'00'000'0;
+
+    u16 sltp_context = [](u8 gr_template) {
+        if (gr_template == 0)
+            return sltp_context_for_template_0;
+        VERIFY(gr_template == 1);
+        return sltp_context_for_template_1;
+    }(inputs.gr_template);
```

Comment:
This suspiciously looks like a
```suggestion
    VERIFY(inputs.gr_template==0 || inputs.gr_template == 1);
    u16 sltp_context = inputs.gr_template ? sltp_context_for_template_1 : sltp_context_for_template_0;
```
which is also used 5 lines higher, so the verify might be redundant, or overdue

Reply:
I think `inputs.gr_template & 1 == inputs.gr_template` might also work but thats way less verbose

Reply:
or as this seems to pop up more often, maybe add an array of  gr_template data to just index once and get everything from

Reply:
Done, thanks. Not sure what I was thinking here.